### ISSUE TITLE
c2st() silently returns NA for $auc because CV folds aren't stratified

### ISSUE THREAD

**Issue by pedroliman** [pedroliman/neuralsbi#269]
## What's wrong

`c2st()`'s docstring (`R/c2st.R`) says the defaults reproduce `sbibm/metrics/c2st.py` exactly, "so a number from here is comparable with a published one." `sbibm` gets its folds from `sklearn.model_selection.cross_val_score(clf, ...)`, which defaults to **stratified** k-fold for a classifier. `c2st()` instead assigns folds with plain, unstratified shuffling:

```r
fold <- base::sample(rep_len(seq_len(n_folds), n))
```

With small or unlucky splits, a fold's test set can end up with zero rows of one class. `roc_auc()` explicitly handles that by returning `NA_real_`:

```r
roc_auc <- function(prob, label) {
  pos <- label == 1
  n1 <- sum(pos)
  n0 <- length(label) - n1
  if (n1 == 0L || n0 == 0L) return(NA_real_)
  ...
}
```

That `NA` then propagates through `mean(aucs)` with no `na.rm` and no warning, so `c2st()$auc` silently comes back `NA` (note: `$accuracy` is unaffected, since accuracy doesn't need both classes present to compute). The function's only size guard (`n_folds >= n_each` check) is written to allow exactly the sample sizes where this bites.

## Repro

```r
library(neuralsbi)
hits <- 0
for (s in 1:200) {
  x <- matrix(rnorm(6 * 2), ncol = 2)
  y <- matrix(rnorm(6 * 2, mean = 3), ncol = 2)
  r <- c2st(x, y, n_folds = 5, seed = s, classifier = "logistic")
  if (any(is.na(r$fold_auc))) hits <- hits + 1
}
cat("NA hits out of 200:", hits, "\n")
```

On this machine: **173 out of 200 seeds** hit at least one `NA` fold AUC at `n_each = 6`, `n_folds = 5`. It's not only a tiny-sample problem: at a more realistic `n_each = 20` it still happens in roughly 3% of seeds over 500 trials.

## Why it matters

Small posterior-sample-count comparisons are an ordinary `c2st()` use case (interactive diagnostics, benchmark sweeps over many seeds/tasks in `inst/benchmarks/`), and `$auc` is one of the two headline numbers the function returns. A `NA` here is silent: no warning is raised, and any code averaging `$auc` across seeds/tasks (as a benchmark sweep would) gets a quietly wrong (`NA`-contaminated) summary rather than an error pointing at the cause.

## Suggested fix

Stratify the fold assignment by label instead of shuffling all rows together — assign folds within each class separately, then union them, matching what `sklearn`'s default stratified k-fold (and hence `sbibm`) actually does. That also restores the doc's "comparable with a published number" claim, since unstratified folds change which rows land in each split relative to `sbibm`'s procedure even when no fold happens to hit the `NA` case.

Found during a code-quality audit; no package code changed as part of filing this.
